wolfcrypt/src/asn.c

```diff
@@ -17310,13 +17315,27 @@ int CompareOcspReqResp(OcspRequest* req, OcspResponse* resp)
         return cmp;
     }
 
-    cmp = req->serialSz - resp->status->serialSz;
-    if (cmp != 0) {
-        WOLFSSL_MSG("\tserialSz mismatch");
-        return cmp;
+    /* match based on found status and return */
+    for (status = resp->status; status; status = next) {
```

Comment:
The function is good for our current use case where we have the same _issuerNameHash_ and  _issuerKeyHash_ but different _serialNumber_. 

Does it make sense to move the for loop at the beginning of the function to handle a general case?

Reply:
`CertStatus` does not contain `issuerHash` or `issuerKeyHash`. The OCSP responses are all for the same key, but different serial numbers. Serial number is all we can match on here.